Document/0x05h-Testing-Platform-Interaction.md

```diff
@@ -804,48 +804,210 @@ Another compliant solution is to define the API level to 17 (JELLY_BEAN_MR1) and
 
 
 
-### Testing Object persistance
+### Testing Object Persistence
 
 #### Overview
 
-There are various ways to persistn an object within Android:
+There are various ways to persist an object within Android:
 
 ##### Object Serialization
 
-An object and its data can be represented as a sequence of bytes. In Java, this is possible using object serialization <sup>[1]</sup>. Serialization is not secure by default and is just a binary format or representation that can be used to store data locally as .ser file. It is possible to sign and encrypt serialized data but, if the source code is available, this is always reversible.  
+An object and its data can be represented as a sequence of bytes. In Java, this is possible using object serialization <sup>[1]</sup>. Serialization is not secure by default and is just a binary format or representation that can be used to store data locally as .ser file. It is possible to encrypt and sign/HMAC serialized data as long as the keys are stored safely. Note that, when classes are changed, the `ObjectInputStream` will not be able to create objects from older .ser files.
```

Comment:
> when classes are changed

I know what you mean, but I think this is not very clear. Maybe put something like this before it: "To deserialize an object, the same version of the class is needed as when it was serialized."